prow/plugins/blockade/blockade.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Example config for this plugin:
+
+blockades:
+- repos:
+	- kubernetes-incubator
+	- kubernetes/kubernetes
+	- kubernetes/test-infra
+  blockregexps:
```

Comment:
document the flow at the top: default allow, deny if a changed file matches any of blockregexps, but allow if it matches any of exception regexps. blocking means adding `do-not-merge/blocked-paths`.